matchtree.go

```diff
@@ -957,18 +957,12 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 				return reposWant[d.repos[docID]]
 			},
 		}, nil
-
-	case *query.Visibility:
-		wantPublic := s.Value == "public"
-		repoVisible := make([]bool, 0, len(d.repoMetaData))
-		for _, r := range d.repoMetaData {
-			repoVisible = append(repoVisible, (r.RawConfig["public"] == "1") == wantPublic)
-		}
+	case *query.RawConfig:
 		return &docMatchTree{
 			reason:  s.String(),
 			numDocs: d.numDocs(),
 			predicate: func(docID uint32) bool {
-				return repoVisible[d.repos[docID]]
+				return s.Encoded&d.repoMetaData[d.repos[docID]].RawConfigEncoded == s.Encoded
```

Comment:
we could get rid of translating into a bitmask and instead calculate the predicate once per repo rather than once per doc.

Reply:
I moved the encoded RawConfig to indexData and consequently removed it from the api. It feels much better now IMO.

I kept everything on document level because otherwise we would have to introduce the assumption that all documents belonging to a repo are stored together in a compound shard. In the future, we might want to tombstone and append single documents which is straightforward to do if we keep everything document based. WDYT?

Reply:
Yeah this looks good to me. BTW what I was thinking was something like

```go
func (d *indexData) repoPredicate(f func(int) bool) func(uint32) bool {
  b := make([]bool, len(d.repoMetaData))
  for i := range b {
    b[i] = f(i)
  }
  return func(docID uint32) bool {
    return b[d.repos[docID]]
  }
}
```

this can then be used in place of docMatchTree predicate

Reply:
Got it. I kept the predicate because I found it a bit easier. We can change it in case we notice an impact on performance.